src/templates/Invoice/InvoiceTemplate.tsx

```diff
@@ -110,15 +151,56 @@ export const InvoiceTemplate: FunctionComponent<TemplateProps<InvoiceDocumentSch
             </div>
             <div className="flex flex-wrap">
               <div className="mb-4 py-2">
-                <p>{billTo?.name}</p>
-                <p>{billTo?.company.name}</p>
-                <p>{billTo?.company.streetAddress}</p>
                 <p>
-                  {billTo?.company.city}
-                  {billTo?.company.postalCode && `, ${billTo?.company.postalCode}`}
+                  <RedactableValue
+                    editable={editable}
+                    value={billTo.name}
+                    onRedactionRequested={() => handleObfuscation(`billTo.name`)}
+                    iconRedact={<IconRedact />}
+                  />
+                </p>
+                <p>
+                  <RedactableValue
+                    editable={editable}
+                    value={billTo.company.name}
+                    onRedactionRequested={() => handleObfuscation(`billTo.company.name`)}
+                    iconRedact={<IconRedact />}
+                  />
+                </p>
+                <p>
+                  <RedactableValue
+                    editable={editable}
+                    value={billTo.company.streetAddress}
+                    onRedactionRequested={() => handleObfuscation(`billTo.company.streetAddress`)}
+                    iconRedact={<IconRedact />}
+                  />
+                </p>
+                <p>
+                  {billTo.company.city}
+                  {billTo.company.postalCode && `, `}
+                  <RedactableValue
+                    editable={editable}
+                    value={billTo.company.postalCode}
+                    onRedactionRequested={() => handleObfuscation(`billTo.company.postalCode`)}
+                    iconRedact={<IconRedact />}
+                  />
+                </p>
+                <p>
+                  <RedactableValue
+                    editable={editable}
+                    value={billTo.company.phoneNumber}
+                    onRedactionRequested={() => handleObfuscation(`billTo.company.phoneNumber`)}
+                    iconRedact={<IconRedact />}
+                  />
+                </p>
+                <p>
+                  <RedactableValue
+                    editable={editable}
+                    value={billTo.email}
+                    onRedactionRequested={() => handleObfuscation(`billTo.email`)}
+                    iconRedact={<IconRedact />}
+                  />
```

Comment:
think u want to map this instead of literal components? (this way is fine also, its explicit)
ie, consolidate data into a array of objects, then map it with redactable value?

Reply:
on second thought, maybe not. due to how comma is added between `city` and `postalCode` keys [here](https://github.com/TradeTrust/generic-templates/pull/49/files#diff-dfae7e911e904b1e9e4b2706d13e74963fcee3af559e22cda92e1b1074011d37R180) + `city` is not to be redacted